### DISCUSSION TITLE
display of an environment variable

### DISCUSSION THREAD

**Discussion by suomiman** [starship/starship#5862] [Q&A] [answered]
hi all

got a a very simple question, that should have a very simple answer, but apparently I'm unable to get this right.
just need to display an env var. 

so made a custom thingie

```
[custom.pass]
command = 'echo "$pass_profile"'
when = true
style = "bg:color_bg3"
format = '[[$symbol($output )](fg:#ffffff bg:color_bg3)]($style)'
```

and the format block looks like

```
format = """
[](color_orange)\
$os\
$username\
[](fg:color_yellow bg:color_aqua)\
$git_branch\
$git_status\
[](fg:color_aqua bg:color_blue)\
$c\
$python\
[](fg:color_blue bg:color_bg3)\
$custom.pass\
[](fg:color_bg3 bg:color_bg1)\
$time\
[ ](fg:color_bg1)\
$line_break\
(fg:color_aqua)\
$directory\
$line_break$character
"""
```

and what I get is this when unset
![unset](https://github.com/starship/starship/assets/36864172/2af64bc0-4e53-4dec-9081-8202fcfc2b73)

and this when set 
![set](https://github.com/starship/starship/assets/36864172/cf2ffc61-b06d-46e1-af53-eca58fec9f1c)

the issues is the string '.pass' in the prompt that should not be there in any case.

obviously doing something wrong here, 
as this should be the easiet bit to display, 
but can't get this right, and been reading the forum way too long, 
so I just thought of asking for help.

Any hints welcome

thanks in advance

**Comment by davidkna** [comment, marked as answer]
Please try using `${custom.pass}` instead. Beyond that, you might want to look at using an [`env_var` module](https://starship.rs/config/#environment-variable) instead of a `custom` module.

**Comment by suomiman** [comment]
thank you!

clearly misunderstood that in the docs.

thanks again!
